### ISSUE TITLE
SC2215 doesn't understand comments after line breaks

### ISSUE THREAD

**Issue by nicknovitski** [koalaman/shellcheck#1104]
#### For bugs
- Rule Id: SC2215
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
pssh --hosts "$(host-inventory)" \
         --askpass \ # TODO: fix key authentication
         --print \
         tail -f /var/log/my.log
```

#### Here's what shellcheck currently says:
```
Line 4:
     --print \
     ^-- SC2215: This flag is used as a command name. Bad line break or missing [ .. ]?
```

#### Here's what I wanted or expected to see:

No errors.  Or at least a different one. 😄

**Comment by Nightfirecat** [comment]
That seems correct. For instance:

```
$ cat test.sh
#!/bin/bash
echo \ # test
abc
$ . test.sh
 # test
bash: abc: command not found
```

**Comment by nicknovitski** [comment]
...How embarrassing!  ShellCheck saves another developer, despite his own best efforts!
